### PR TITLE
feat(reso): new agent for REUSE.Software standard

### LINKED ISSUE #1833: Detect 'foobar.license' files, as part of RESUE.Software standard
### Description

Following discussion in #1592, this issue proposes a first step.

The idea would be to support the `file.license` way of declaring a license and copyright for binary files (or any file in which compliance text cannot be inserted, like `json`or `yaml` files)

Extract of the specification: https://reuse.software/spec/#comment-headers
> If a file is not a plain text file or does not permit the inclusion of comments, the comment header that declares the file’s Copyright and Licensing Information SHOULD be in an adjacent file of the same name with the additional extension .license (example: cat.jpg.license if the original file is cat.jpg).

I am not sure if the mechanism should be a new agent, or be implemented inside the existing Ojo agent.
I tend to like the idea of reusing Ojo since:
- The `file.license` files will contain SPDX tags
- The files for which compliance is described with a `file.license` file can be safely identified automatically, and this is the spirit of the Ojo agent.

I am not sure, either if the Copyright should be handled together, or separately by the Copyright agent.
If it should be handled separately, then I suggest that this issue concentrates on the License, and we will see the Copyright in an other issue.

#### Steps followed and expected result

Upload an archive containing:
```
config.json
config.json.license
volcano.jpg
voclano.jpg.license
```

The files `*.license` both contain someting like:
```
# SPDX-FileCopyrightText: 2016, 2018-2019 Jane Doe <jane@example.com>
# SPDX-FileCopyrightText: 2019 Example Company
#
# SPDX-License-Identifier: GPL-3.0-or-later
```

Then Fossology would  identify all 4 files as `GPL-3.0-or-later` automatically (and possibly the Copyrights too).

### PR DESCRIPTION
**Pull request by NicolasToussaint**
Replaces PR: https://github.com/fossology/fossology/pull/2029

Fixes #1833

### CODE REVIEW

File: src/www/ui/api/Models/Analysis.php [hunk 1]
```diff
@@ -69,6 +70,11 @@ class Analysis
    * Whether to schedule ojo agent or not
    */
   private $ojo;
+  /**
+   * @var boolean $pkgagent
```
Comment (GMishx) [inline src/www/ui/api/Models/Analysis.php:74]: Oops, the doc was updated for the variable.

**Review by GMishx — APPROVED** [review]
Changes looks good. Just one documentation needs update.

**Review by GMishx — APPROVED** [review]
Tested, working as expected.

File: src/www/ui/api/Models/Analysis.php (hunk 1 above)
Comment (NicolasToussaint) [reply src/www/ui/api/Models/Analysis.php:74]: well spotted ;-)

**Pull request merged**